nixos/modules/services/web-apps/wordpress.nix

```diff
@@ -106,62 +106,36 @@ let
         };
 
         plugins = mkOption {
-          type = types.listOf types.path;
-          default = [];
+          type = types.attrsOf types.path;
+          default = {};
```

Comment:
This would break existing configurations. Maybe something like
```nix
with types; coercedTo
  (listOf path)
  (l: warn "setting this option with a list is deprecated"
    listToAttrs (map (p: nameValuePair (p.name or throw "${p} does not have a name") p) l))
  (attrsOf path)
```
@roberth any thoughts?

Reply:
my thought was that this would intentionally be a breaking change introduced in `23.05`, so that we do not need to maintain a bodge like #210676 or your suggestion in perpetuity; but I am amenable to any solution that resolves my core issue.

Reply:
If a third party provides wordpress modules, they'd have to pick a NixOS version to support, which is not great.
`coercedTo` sounds good, especially with the warning. That way the list support can be removed after some time - no perpetuity.

Reply:
to be clear, when I say "third party", I mean the the collection of "unofficial wordpress plugins" that we [recte I] maintain: https://github.com/NixOS/nixpkgs/blob/master/pkgs/servers/web-apps/wordpress/packages/thirdparty.nix

Reply:
before applying anything I would like some consensus and an applyable suggestion, as I am quite new to the nixos module typesystem; happy to take whatever approach you all deem reasonable for implementing the breaking change. (also let me know how/what can be backported, since it seems like @ncfavier's suggestion is permissible for 22.11.)